site/docs/spec.md

```diff
@@ -646,6 +647,25 @@ Each partition field in the fields list is stored as an object. See the table fo
 
 In some cases partition specs are stored using only the field list instead of the object format that includes the spec ID, like the deprecated `partition-spec` field in table metadata. The object format should be used unless otherwise noted in this spec.
 
+#### Partition Field ID Handling
```

Comment:
This is in the appendix that covers JSON serialization for partition specs. It is good to document the `field-id` name here and note that it is required in v2 and optional in v1, but a lot of this content should be noted in the [Partitioning](http://iceberg.apache.org/spec/#partitioning) section of the spec.